NDXINTER/reduce.py

```diff
@@ -1,41 +1,171 @@
+
+    # This looks for the run angle and set other parameters accordingly
+    # Using a tolerance of +-0.5% of the motor readback angle
+    min = angle * 0.995
+    max = angle * 1.005
+    angle_found = False
+    for i in rows:
+        # If the value is within -0.5% to +0.5% it is counted as a match
+        if min <= float(i[0]) <= max:
+            angle_found = True
+            first_transmission_run_list = i[1]
+            second_transmission_run_list = i[2]
+            transmission_processing_instructions = i[3]
+            # Skipping over parameters that are present in the JSON file but not currently used in the reduction
+            processing_instructions = i[8]
+            break
+
+    # This is the default case
+    if not angle_found:
+        for i in rows:
+            if i[0] == "":
+                angle_found = True
+                first_transmission_run_list = i[1]
+                second_transmission_run_list = i[2]
+                transmission_processing_instructions = i[3]
+                # Skipping over parameters that are present in the JSON file but not currently used in the reduction
+                processing_instructions = i[8]
+                break
+
+    if not angle_found:
+        raise Exception # Excpetion for if neither a pre-defined angle nor the default case are found
+
+    start_overlap = experimentView["startOverlapEdit"]
+    end_overlap = experimentView["endOverlapEdit"]
+
+    #========================================================================================
+    # Instrument Settings
+    #========================================================================================
+
+    instrumentView = data["instrumentView"]
+
+    monitor_integration_wavelength_min = instrumentView["monIntMinEdit"]
+    monitor_integration_wavelength_max = instrumentView["monIntMaxEdit"]
+    monitor_background_wavelength_min = instrumentView["monBgMinEdit"]
+    monitor_background_wavelength_max = instrumentView["monBgMaxEdit"]
+    wavelength_min = instrumentView["lamMinEdit"]
+    wavelength_max = instrumentView["lamMaxEdit"]
+    i_zero_monitor_index = instrumentView["I0MonitorIndex"]
+
+    # Set a string based on what integer value is found
+    if instrumentView["detectorCorrectionTypeComboBox"] == 1:
+        detector_correction_type = "RotateAroundSample"
+    elif instrumentView["detectorCorrectionTypeComboBox"] == 0:
+        detector_correction_type = "VerticalShift"
+    else:
+        raise Exception # If the value isn't 1 or 0 then it isn't valid
+
+    return analysis_mode, first_transmission_run_list, second_transmission_run_list, transmission_processing_instructions, \
+    processing_instructions, start_overlap, end_overlap, monitor_integration_wavelength_min, \
+    monitor_integration_wavelength_max, monitor_background_wavelength_min, monitor_background_wavelength_max, wavelength_min, \
+    wavelength_max, i_zero_monitor_index,detector_correction_type

```

Comment:
This might be easier to manage if you make a [dataclass](https://docs.python.org/3/library/dataclasses.html), something like

```
from dataclasses import dataclass

@dataclass
class INTERParams:
    analysis_mode
    first_transmission_run_list
    second_transmission_run_list
    transmission_processing_instructions
    processing_instructions
    start_overlap
    end_overlap
    monitor_integration_wavelength_min
    monitor_integration_wavelength_max
    monitor_background_wavelength_min
    monitor_background_wavelength_max
    wavelength_min
    wavelength_max: float
    i_zero_monitor_index
    detector_correction_type
```
that also allows you to set typed, e.g. `wavelength_max: float`

Then just make it at the start of the function `params = INTERParams()` and fill it with data as you go down